addon/components/plugins/text-style/color.ts

```diff
@@ -0,0 +1,98 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { SayController } from '@lblod/ember-rdfa-editor';
+
+import IntlService from 'ember-intl/services/intl';
+import { TextSelection } from 'prosemirror-state';
+
+type Args = {
+  controller?: SayController;
+};
+export default class Color extends Component<Args> {
+  @service declare intl: IntlService;
+  colors = ['black', 'red', 'green', 'blue', 'orange'];
+
+  get controller() {
+    return this.args.controller;
+  }
+
+  get currentStyle() {
+    const currentColor = this.colors.find(this.colorIsActive);
```

Comment:
you calculate the active color in the document
then you loop over the color array, comparing each color to the active color, to find... the active color?

Reply:
yes, you are right. In this commit I was mainly interested if functionality is working as expected, and for that reason forget this. You are right, doesn't make sense, sorry for that.